dlt/destinations/job_client_impl.py

```diff
@@ -431,10 +433,17 @@ def _commit_schema_update(self, schema: Schema, schema_str: str) -> None:
 
 
 class SqlJobClientWithStaging(SqlJobClientBase, WithStagingDataset):
+
+    in_staging_mode: bool = False
+
     @contextlib.contextmanager
     def with_staging_dataset(self)-> Iterator["SqlJobClientBase"]:
-        with self.sql_client.with_staging_dataset(True):
-            yield self
+        try:
```

Comment:
@rudolfix in athena client i need to know wether it is being run in staging mode. This breaks this nice whole idea of the client being (mostly) agnostic of the underlaying dataset, but I'm not sure how to do it otherwise.